contracts/stake-cw20-external-rewards/src/contract.rs

```diff
@@ -62,6 +62,10 @@ pub fn instantiate(
     };
     CONFIG.save(deps.storage, &config)?;
 
+    if msg.reward_duration == 0 {
```

Comment:
Maybe this validation could be moved to the top of the function? It would allow avoiding doing a bunch of unnecessary setup when 0 is passed.

Reply:
This is a really interesting point. I think I'm inclined to keep it the way it is for two reasons:

1. Anything could fail first. Changing the order likely only makes sense if we have some heuristic for deciding which error condition is going to fail more frequently.
2. IIRC (@de-husk knows the details on this) in CosmWasm your gas payment is based on gas wanted and not gas used. So, if I budget N gas for this instantiation and it fails early because we reorganize these and uses M gas, I sill pay N gas for the execution.

Appreciate you taking a close look at this and the good feedback! Happy to be convinced otherwise on either of these points.

Reply:
Totally correct about paying for GasWanted, not GasUsed here. 

However, in practice `GasWanted` is usually set automatically by calling the `/simulate` cosmos abci query, and I need to learn more about how cosmwasm gas simulation is done, but it runs a read-only (single node / non-consensus) dry-run of the code using stale data and returns the gas used. So I think you should actually be lowering the gas threshold by exiting early for most people.